app/models/blueprint.rb

```diff
@@ -52,6 +52,8 @@ def copy_service_template(blueprint, service_template, is_top = false, new_attri
       duplicate_resource_actions(service_template, new_template, is_top)
       duplicate_service_resources(blueprint, service_template, new_template)
       update_copied_service_template(blueprint, new_template, new_attributes)
+      duplicate_custom_buttons(service_template, new_template)
+      duplicate_custom_button_sets(service_template, new_template)
```

Comment:
@bdunne In general I want `update_copied_service_template` to be the last line because `new_attributes` should be set to overwrite any copied data. If your copy method requires new template to be saved first, you can refactor.